ci-operator/step-registry/cucushift/installer/check/vsphere/cpms/cucushift-installer-check-vsphere-cpms-commands.sh

```diff
@@ -0,0 +1,83 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+export KUBECONFIG=${SHARED_DIR}/kubeconfig
+
+
+
+function checkMultizone() {
+    check_result=0
+    cpms_network=$(oc get controlplanemachineset -n openshift-machine-api -ojson | jq -r '.items[].spec.template.machines_v1beta1_machine_openshift_io.spec.providerSpec.value.network.devices')
+    cpms_workspace=$(oc get controlplanemachineset -n openshift-machine-api -ojson | jq -r '.items[].spec.template.machines_v1beta1_machine_openshift_io.spec.providerSpec.value.workspace')
+    cpms_template=$(oc get controlplanemachineset -n openshift-machine-api -ojson | jq -r '.items[].spec.template.machines_v1beta1_machine_openshift_io.spec.providerSpec.value.template')
+#check network,workspace and template field in cpms.
+    if [[ ${cpms_network} == "null" ]] && [[ ${cpms_workspace} == "{}" ]] && [[ ${cpms_template} == "" ]]; then
+        echo "INFO: For multi failure domains The network,workspace and template under cpms are empty, that's expected"
+    else
+	echo "ERROR: For multi failure domains The network,workspace and template under cpms should be empty. network is ${cpms_network},workspace is ${cpms_workspace},template is ${cpms_template}" 
+        check_result=1
+    fi
+    
+#check the name of failuredomains in cpms should be same with infrastructure.
+    fd_name_cpms=$(oc get controlplanemachineset -n openshift-machine-api -ojson | jq -r '.items[].spec.template.machines_v1beta1_machine_openshift_io.failureDomains.vsphere[].name')
+    fd_name_infra=$(oc get infrastructure -n openshift-machine-api -ojson | jq -r '.items[].spec.platformSpec.vsphere.failureDomains[].name' | head -n3)
```

Comment:
What is the output of `oc get infrastructure -n openshift-machine-api -ojson | jq -r '.items[].spec.platformSpec.vsphere.failureDomains[].name'` ? How can we decide it is the 3rd one (`head -n3`) ?

Reply:
oc get infrastructure -n openshift-machine-api -ojson | jq -r '.items[].spec.platformSpec.vsphere.failureDomains[].name'
us-east-1
us-east-2
us-east-3
us-west-1
head -n3 display 3 failure domains.  the last failure domain is for work node.

Reply:
can you show me the whole output of `oc get infrastructure -n openshift-machine-api -ojson`

Reply:
From `ipi-conf-vsphere-zones`, the install-config.yaml looks like:
```
controlPlane:
  name: "master"
  replicas: 3
  platform:
    vsphere:
      zones:
       - "us-east-1"
       - "us-east-2"
       - "us-east-3"
compute:
- name: "worker"
  replicas: 4
  platform:
    vsphere:
      zones:
       - "us-east-1"
       - "us-east-2"
       - "us-east-3"
       - "us-west-1"
```
So I think the `head -n3` is not stable enough, we should get the expected value from install-config.yaml, just like what check-azure-cpms did.

Reply:
> oc get infrastructure -n openshift-machine-api -ojson

 oc get infrastructure -n openshift-machine-api -ojson                                                                                
{
    "apiVersion": "v1",
    "items": [
        {
            "apiVersion": "config.openshift.io/v1",
            "kind": "Infrastructure",
            "metadata": {
                "creationTimestamp": "2024-02-01T10:08:05Z",
                "generation": 1,
                "name": "cluster",
                "resourceVersion": "519",
                "uid": "8ff874ee-6e91-4a75-8efb-6b4538eb87d7"
            },
            "spec": {
                "cloudConfig": {
                    "key": "config",
                    "name": "cloud-provider-config"
                },
                "platformSpec": {
                    "type": "VSphere",
                    "vsphere": {
                        "failureDomains": [
                            {
                                "name": "us-east-1",
                                "region": "us-east",
                                "server": "vcs8e-vc.ocp2.dev.cluster.com",
                                "topology": {
                                    "computeCluster": "/IBMCloud/host/vcs-mdcnc-workload-1",
                                    "datacenter": "IBMCloud",
                                    "datastore": "/IBMCloud/datastore/mdcnc-ds-shared",
                                    "networks": [
                                        "ci-vlan-1287"
                                    ],
                                    "resourcePool": "/IBMCloud/host/vcs-mdcnc-workload-1//Resources"
                                },
                                "zone": "us-east-1a"
                            },
                            {
                                "name": "us-east-2",
                                "region": "us-east",
                                "server": "vcs8e-vc.ocp2.dev.cluster.com",
                                "topology": {
                                    "computeCluster": "/IBMCloud/host/vcs-mdcnc-workload-2",
                                    "datacenter": "IBMCloud",
                                    "datastore": "/IBMCloud/datastore/mdcnc-ds-shared",
                                    "networks": [
                                        "ci-vlan-1289"
                                    ],
                                    "resourcePool": "/IBMCloud/host/vcs-mdcnc-workload-2//Resources"
                                },
                                "zone": "us-east-2a"
                            },
                            {
                                "name": "us-east-3",
                                "region": "us-east",
                                "server": "vcs8e-vc.ocp2.dev.cluster.com",
                                "topology": {
                                    "computeCluster": "/IBMCloud/host/vcs-mdcnc-workload-3",
                                    "datacenter": "IBMCloud",
                                    "datastore": "/IBMCloud/datastore/mdcnc-ds-shared",
                                    "networks": [
                                        "ci-vlan-1300"
                                    ],
                                    "resourcePool": "/IBMCloud/host/vcs-mdcnc-workload-3//Resources"
                                },
                                "zone": "us-east-3a"
                            },
                            {
                                "name": "us-west-1",
                                "region": "us-west",
                                "server": "vcs8e-vc.ocp2.dev.cluster.com",
                                "topology": {
                                    "computeCluster": "/datacenter-2/host/vcs-mdcnc-workload-4",
                                    "datacenter": "datacenter-2",
                                    "datastore": "/datacenter-2/datastore/mdcnc-ds-shared",
                                    "networks": [
                                        "ci-vlan-1287"
                                    ],
                                    "resourcePool": "/datacenter-2/host/vcs-mdcnc-workload-4//Resources"
                                },
                                "zone": "us-west-1a"
                            }
                        ],
                        "nodeNetworking": {
                            "external": {},
                            "internal": {}
                        },
                        "vcenters": [
                            {
                                "datacenters": [
                                    "IBMCloud"
                                ],
                                "port": 443,
                                "server": "vcs8e-vc.ocp2.dev.cluster.com"
                            }
                        ]
                    }
                }
            },
            "status": {
                "apiServerInternalURI": "https://api-int.wweitest1.qe.devcluster.openshift.com:6443",
                "apiServerURL": "https://api.wweitest1.qe.devcluster.openshift.com:6443",
                "controlPlaneTopology": "HighlyAvailable",
                "cpuPartitioning": "None",
                "etcdDiscoveryDomain": "",
                "infrastructureName": "wweitest1-ldkpq",
                "infrastructureTopology": "HighlyAvailable",
                "platform": "VSphere",
                "platformStatus": {
                    "type": "VSphere",
                    "vsphere": {
                        "apiServerInternalIP": "10.38.201.137",
                        "apiServerInternalIPs": [
                            "10.38.201.137"
                        ],
                        "ingressIP": "10.38.201.138",
                        "ingressIPs": [
                            "10.38.201.138"
                        ]
                    }
                }
            }
        }
    ],
    "kind": "List",
    "metadata": {
        "resourceVersion": ""
    }
}

Reply:
> From `ipi-conf-vsphere-zones`, the install-config.yaml looks like:
> 
> ```
> controlPlane:
>   name: "master"
>   replicas: 3
>   platform:
>     vsphere:
>       zones:
>        - "us-east-1"
>        - "us-east-2"
>        - "us-east-3"
> compute:
> - name: "worker"
>   replicas: 4
>   platform:
>     vsphere:
>       zones:
>        - "us-east-1"
>        - "us-east-2"
>        - "us-east-3"
>        - "us-west-1"
> ```
> 
> Similar to [#48371 (comment)](https://github.com/openshift/release/pull/48371#discussion_r1477239445), we should not rely on cluster return to check itself, but should from install-config.yaml. So I think the `head -n3` is not stable enough, we should get the expected value from install-config.yaml, just like what check-azure-cpms did.

ok. updated